app/controllers/admin/admin_controller.rb

```diff
@@ -1,16 +1,26 @@
 class Admin::AdminController < ApplicationController
+  # include UsersHelper # Perhaps need to create the same method in AdminHelper and include it?
+
   def index
     @users = User.find(:all, :conditions => ["admin = ?", false])
     authorize current_user
   end
 
   def show
-    @user = User.find(params[:id])
+    @user = User.find_by_id(params[:id])
+    if @user.nil?
+      redirect_to admin_root_path
+      return
+    end
     authorize current_user
   end
 
   def edit
-    @user = User.find(params[:id])
+    @user = User.find_by_id(params[:id])
+    if @user.nil?
+      redirect_to admin_root_path
+      return
+    end
     authorize current_user
```

Comment:
For all these authorize current_user, why not make it in a after filter? like :after authorize_current_session. and have authorize current_user in a func.